os-server/src/main/java/app/openschool/discussion/DiscussionAnswerMentorServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package app.openschool.discussion;
+
+import app.openschool.common.exceptionhandler.exception.PermissionDeniedException;
+import app.openschool.discussion.dto.DiscussionAnswerRequestDto;
+import app.openschool.discussion.dto.DiscussionAnswerResponseDto;
+import app.openschool.discussion.mapper.DiscussionAnswerMentorMapper;
+import app.openschool.user.User;
+import app.openschool.user.UserRepository;
+import java.security.Principal;
+import java.time.Instant;
+import org.springframework.stereotype.Service;
+
+
+@Service("discussionAnswerMentor")
+public class DiscussionAnswerMentorServiceImpl implements DiscussionAnswerService {
```

Comment:
you don't need to create separate services for each entity, the propose of the service is to work with different repositories combine the results, make some orchestration etc. 
here we can have PeersDiscussionService and MentorDiscussionService and that's it. all question answers will be there

Reply:
I have only two services , only with wrong names.